R/getNeatOrder.R

```diff
@@ -0,0 +1,173 @@
+#' centers the data using specified methods (\code{mean}, \code{median}, or \code{none}), and then calculates 
+#' the angle (theta) for each point relative to the centroid. The data points are then 
+#' sorted based on these theta values in ascending order. 
+#' 
+#' One significant application of this sorting method is in plotting heatmaps. 
+#' By using radial theta sorting, the relationships between data points can be preserved 
+#' according to the ordination method's spatial configuration, rather than relying on 
+#' hierarchical clustering, which may distort these relationships. This approach 
+#' allows for a more faithful representation of the data's intrinsic structure as captured 
+#' by the ordination process.
+#' 
+#' @param x A matrix containing the ordinated data to be sorted. Columns should represent the principal components (PCs) and rows should represent the entities being analyzed (e.g. features or samples). There should be 2 columns only representing 2 PCs.
+#' @param centering A single \code{character} value specifying the method to center the data. Options are \code{mean}, \code{median}, or \code{none} if your data is already centered. (default: method = \code{mean})
+#' @param ... Additional arguments passed to other methods.
+#' @return A \code{character} vector of row names in the sorted order.
+#' 
+#' @details 
+#' It's important to note that the \pkg{sechm} package does actually have the functionality for plotting a heatmap using this radial theta angle ordering, though only by using an MDS ordination. 
+#' 
+#' This functionality can be found at:
+#' 
+#'  \url{https://bioconductor.org/packages/3.18/bioc/vignettes/sechm/inst/doc/sechm.html#row-ordering}. 
+#'  
+#' That being said, the \code{getNeatOrder} function is more modular and separate to the plotting, and 
+#' can be applied to any kind of ordinated data which can be valuable depending on the use case.
+#' 
+#' The below paper outlines the NeatMap method in more detail:
+#' 
+#' "Rajaram & Oono (2010)" - https://doi.org/10.1186/1471-2105-11-45.
+#' 
+#' @name getNeatOrder
+#' 
+#' @examples
+#' ## Load the required libraries and dataset
+#' library(mia)
+#' library(scater)
+#' library(ComplexHeatmap)
+#' library(circlize)
+#' data(peerj13075)
+#' 
+#' ## Group data by taxonomic order
+#' tse <- agglomerateByRank(peerj13075, rank = "order", onRankOnly = TRUE)
+#' 
+#' ## Add a pseudocount to the counts data
+#' assay(tse, "counts") <- assay(tse, "counts") + 1
+#' 
+#' ## Transform the samples into relative abundances using CLR
+#' tse <- transformAssay(tse, assay.type = "counts", method="clr", MARGIN = "samples", name="clr")
+#' 
+#' ## Transform the features (taxa) into zero mean, unit variance (z transformation)
+#' tse <- transformAssay(tse, assay.type="clr", method="z", MARGIN = "features", name="z")
+#' 
+#' ## Perform PCA using calculatePCA
+#' reducedDim(tse, "PCA") <- calculatePCA(tse, ncomponents = 10, assay.type = "z")
```

Comment:
You can skip this part. Just get the result to new variable. The idea was to make the example simpler.

Reply:
Also ncomponents is unnecessary here, I guess

Reply:
res <- calculatePCA(tse, assay.type = "z")
order <- getNeatOrder(res, ...)

Reply:
OK will do, yeah I can understand that this is abit simpler to the user, but then again I also think it's just personal preference at a certain point. Cause I code in Rust a fair bit you have to manage memory quite a lot, so I generally avoid creating excess variable in my code, as this is more memory efficient, so for me I find saving stuff to an object simpler than creating excess variables.

Reply:
Yeah sure I can change ncomponents, although then again it's just like if you were doing this in reality, the default is 500 components, so you would probably cap the ncomponents at a certain point as you just dont want to deal with that many components

Reply:
To store the result to TreeSE, you can use runPCA. However, as these are examples of usage of getNeatOrder(), we should make the code that is not directly related to this function as simple as possible

For instance, we have found that reducedDim might be hard to understand first, so it is better to make the examples simple so it is easy for all users to see what is happening.

Reply:
One point of using TreeSE is the memory management. However, OMA is more suitable for more detailed information and examples  (including best practices)

Reply:
ok sure will do, I get you

Reply:
Was that the only change by the way, making those changes quickly now, lmk if there's anything else to change, I can't really see anything else with the code but there may be small things you notice I don't